### PR TITLE
auto-attach: use api full_auto_attach

### PR DESCRIPTION
**Pull request by orndorffgrant**
## Test Steps

Run auto-attach tests to make sure they work as expected

## Desired commit type

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist

 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: uaclient/actions.py [hunk 1]
```diff
@@ -249,9 +249,6 @@ def get_cloud_instance(
     try:
         instance = identity.cloud_instance_factory()
     except exceptions.CloudFactoryError as e:
-        if cfg.is_attached:
-            # We are attached on non-Pro Image, just report already attached
-            raise exceptions.AlreadyAttachedError(cfg)
```
Comment (lucasmoura) [inline uaclient/actions.py:254]: Why are we removing this bit here ?

File: uaclient/api/u/pro/attach/auto/full_auto_attach/v1.py [hunk 2]
```diff
@@ -66,9 +66,29 @@ def full_auto_attach(options: FullAutoAttachOptions) -> FullAutoAttachResult:
     return _full_auto_attach(options, UAConfig(root_mode=True))
 
 
-def _full_auto_attach(options: FullAutoAttachOptions, cfg: UAConfig):
+def _full_auto_attach(
```
Comment (lucasmoura) [inline uaclient/api/u/pro/attach/auto/full_auto_attach/v1.py:69]: Should we add an unittest where we fail to acquire the lock ?

Comment (orndorffgrant) [reply uaclient/api/u/pro/attach/auto/full_auto_attach/v1.py:69]: Good idea!

File: uaclient/actions.py (hunk 1 above)
Comment (orndorffgrant) [reply uaclient/actions.py:254]: This check already happens for cli auto-attach at cli.py:1273 and for the api, this PR adds the check at full_auto_attach/v1.py:89

**Review by lucasmoura — APPROVED** [review]
LGTM, just running again the failed CI jobs

**Review by renanrodrigo — APPROVED** [review]
lgtm

**Pull request merged**